### ISSUE TITLE
Remove dependence on Chrome Driver for API integration tests.

### ISSUE THREAD

**Issue by kdlucas** [google/cloudprint_logocert#6]
Where possible, remove dependence on ChromeDriver and interact directly with the GCP APIs directly. This will eliminate test brittleness especially when UI elements change.

**Comment by kdlucas** [comment]
Creating a new file _cpslib.py, which contains a new class GCPService that will interact directly with the GCP backend APIs.

**Comment by kdlucas** [comment]
Ok, so I have most of the methods done for this, which speaks to the Cloud Print Service directly and doesn't use Chrome and Webdriver to make queries. The only part left is to create print jobs using the API, and not Chrome. If anyone wants to take this on, just take this issue and then use the cpslib branch that I created.

**Comment by bryangl** [comment]
Hi Kelly, thanks, this will speed things up quite a bit.

**Comment by kdlucas** [comment]
Yashiro-san, is this something you can do? I got it working except I'm not sure how to send a print job correctly using only the API. If you know how you might be able to migrate to using the Cloud Print Service API's and then we could bypass the fragility of depending on Webdriver and the UI to run print jobs.

**Comment by YasuhiroFuruta** [comment]
I know how to print via local / cloud printing using APIs. I implemented local and cloud printing test application using Java few years ago.

IMPORTANT: The acceptable file format is very limited (e.g. PWG-raster only) on local printing API. Chrome converts the file format for local printing.

**Comment by kdlucas** [comment]
Well, the hard part is figuring out all the parameters for the print job,
so if you can figure that out then just look at my cpslib branch, and add a
couple of methods for printing.

Kelly
kdLucas

**Comment by kdlucas** [comment]
And we'll need to add printing methods for both local print and for cloud printing.

**Comment by kdlucas** [comment]
Kevin Cho is now taking this on, and has completed most of the heavy lifting, so we should see this land to the master branch before 2017.

**Comment by kdlucas** [comment]
I'm going to test the logocert_api branch on OS X on Thursday, so we'll see how it goes. I'll file any issues against it that I find.

**Comment by lrbom** [comment]
Awesome, thanks Kelly. Look forward to hearing your feedback!

**Comment by kdlucas** [comment]
The logocert_api branch has now been tested. We're waiting for a few other key persons to verify this branch, and then it will be merged into the main branch. Once logocert_api is merged into the main branch, this issue should be closed.

**Comment by vamshikk** [comment]
Completed.
